compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -867,14 +873,15 @@ private BType getMatchingReadonlyIntersectionBoundType(BIntersectionType interse
 
     private BTupleType getMatchingTupleBoundType(BTupleType expType, SymbolEnv env, HashSet<BType> resolvedTypes) {
         boolean hasDifferentType = false;
-        List<BType> tupleTypes = new ArrayList<>();
-        for (BType type : expType.tupleTypes) {
-            BType matchingBoundType = getMatchingBoundType(type, env, resolvedTypes);
-            if (!hasDifferentType && isDifferentTypes(type, matchingBoundType)) {
+        List<BTupleMember> tupleTypes = new ArrayList<>();
+        for (BTupleMember type : expType.memberTypes) {
```

Comment:
Why aren't we using `getTupleTypes()`? Please check and update wherever applicable.